scipy/spatial/tests/test_spherical_voronoi.py

```diff
@@ -12,11 +12,70 @@
 from scipy.spatial import _spherical_voronoi as spherical_voronoi
 from scipy.spatial.transform import Rotation
 from scipy.optimize import linear_sum_assignment
+from scipy.constants import golden as phi
+from scipy.spatial import geometric_slerp
 
 
 TOL = 1E-10
 
 
+def _generate_tetrahedron():
+    return np.array([[1, 1, 1], [1, -1, -1], [-1, 1, -1], [-1, -1, 1]])
+
+
+def _generate_cube():
+    return np.array(list(itertools.product([-1, 1.], repeat=3)))
+
+
+def _generate_octahedron():
+    return np.array([[-1, 0, 0], [+1, 0, 0], [0, -1, 0],
+                     [0, +1, 0], [0, 0, -1], [0, 0, +1]])
+
+
+def _generate_dodecahedron():
+
+    x1 = _generate_cube()
+    x2 = np.array([[0, -phi, -1 / phi],
+                   [0, -phi, +1 / phi],
+                   [0, +phi, -1 / phi],
+                   [0, +phi, +1 / phi]])
+    x3 = np.array([[-1 / phi, 0, -phi],
+                   [+1 / phi, 0, -phi],
+                   [-1 / phi, 0, +phi],
+                   [+1 / phi, 0, +phi]])
+    x4 = np.array([[-phi, -1 / phi, 0],
+                   [-phi, +1 / phi, 0],
+                   [+phi, -1 / phi, 0],
+                   [+phi, +1 / phi, 0]])
+    return np.concatenate((x1, x2, x3, x4))
+
+
+def _generate_icosahedron():
+    x = np.array([[0, -1, -phi],
+                  [0, -1, +phi],
+                  [0, +1, -phi],
+                  [0, +1, +phi]])
+    return np.concatenate([np.roll(x, i, axis=1) for i in range(3)])
+
+
+def _generate_polyhedron(name):
+
+    if name == "tetrahedron":
+        p = _generate_tetrahedron()
+    elif name == "cube":
+        p = _generate_cube()
+    elif name == "octahedron":
+        p = _generate_octahedron()
+    elif name == "dodecahedron":
+        p = _generate_dodecahedron()
+    elif name == "icosahedron":
+        p = _generate_icosahedron()
+    else:
+        raise ValueError("unrecognized polyhedron")
```

Comment:
This is the first line of code in SphericalVoronoi source or testing that isn't covered by a test. Since it is a test support function I didn't insist it should be covered, because then you're testing your tests, but I also wasn't sure the error handling was called for on a test support function & it can be nice to have 100 % coverage across the board.

Minor point though.

Reply:
I generate some polyhedra in spatial/transform/tests/test_rotation_groups.py` too, so there is some duplication of functionality here. We could collect the functions in a separate file if you think it is appropriate.